### ISSUE TITLE
Support Flatcar 

### ISSUE THREAD

**Issue by mcbenjemaa** [ionos-cloud/cluster-api-provider-proxmox#302]
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

To support Flatcar, we must ensure that Proxmox can load the ignition data into the VM.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

we have a way of injecting an ISO into the VM; while providing ignition data.

**Environment:**

- Cluster-api-provider-proxmox version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

**Comment by mcbenjemaa** [comment]
@tormath1 yes, I know that,
However, I'm also trying to do some hacks by injecting an ISO as config-2 with the OpenStack provider.
don't know if that works though.

**Comment by mcbenjemaa** [comment]
networkd config should be adjusted so it can support our current features:

Ref: https://www.freedesktop.org/software/systemd/man/latest/systemd.network.html

**Comment by vxav** [comment]
Hi @mcbenjemaa and thanks for the flatcar PR!

I tried it and I see the iso gets mounted to the node VM. 
The contents of the ISO look ok (by mounting it to the pve host).
I'm using CAPMOX `0.6.2` and a Flatcar image built with `image-builder`.

However it doesn't appear to be in the guest OS of the node VM. Ignition gets no data and `/etc/hostname` and `/etc/proxmox-env` (which I assume are the metadata) don't exist.

Any idea what could be missing here?

**Comment by mcbenjemaa** [comment]
@vxav 
Are you using the correct cluster template?
And did you enabled bootstrap format ignition?

**Comment by vxav** [comment]
Yep I used the Flatcar flavor which sets the correct flags (format ignition included). The [template](https://gist.github.com/vxav/a59edf7bc985ccae64dd5fc56b5a96f2#proxmox-flatcar-template) looks right.

**Comment by mcbenjemaa** [comment]
Please check the journalctl if there is anything wrong with ignition,
and also check if the afterburn runs successfully!

**Comment by vxav** [comment]
Well ignition doesn't any source of metadata ([log](https://gist.github.com/vxav/a59edf7bc985ccae64dd5fc56b5a96f2#ignition-logs)).

Do you have a proxmox flatcar image pushed somewhere? (S3 or else). I'd think my image is right but I don't really know what else it could be here.
I'm not familiar with afterburn so not sure where to look here but I noticed that /run/ignition.env is set to qemu.

```
localhost /home/builder # cat /run/ignition.env
OEM_ID=qemu
PLATFORM_ID=qemu
```

Shouldn't it be [proxmoxve](https://coreos.github.io/ignition/supported-platforms/)?

**Comment by mcbenjemaa** [comment]
Exactly, you should set OEM_ID proxmoxve

**Comment by vxav** [comment]
I used the new proxmox-flatcar image-builder flavor with `make build-proxmox-flatcar`. 
Did you get the correct ID when you used it? Or did you build your images differently?

**Comment by vxav** [comment]
I think I see what I missed here https://github.com/kubernetes-sigs/image-builder/blob/main/images/capi/packer/proxmox/README.md#flatcar-for-proxmox

```
## flatcar version
export FLATCAR_VERSION=4152.1.0
export FLATCAR_CHANNEL=beta

export OEM_ID=proxmoxve # make sure to choose OEM_ID=proxmoxve
```
